base/utf16.jl

```diff
@@ -1,10 +1,82 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
+#=
+@doc """
+@brief      Error messages for Unicode / UTF support
+""" ->
+=#
+#=
+@enum(UTF_ERR,
+      SHORT = 1,
+      CONT,
+      LONG,
+      NOT_LEAD,
+      NOT_TRAIL,
+      NOT_SURROGATE,
+      MISSING_SURROGATE,
+      INVALID,
+      SURROGATE,
+      NULL_TERMINATE,
+)
+=#
+
+const UTF_ERR_SHORT = 1
+const UTF_ERR_CONT  = 2
+const UTF_ERR_LONG  = 3
+const UTF_ERR_NOT_LEAD = 4
+const UTF_ERR_NOT_TRAIL = 5
+const UTF_ERR_NOT_SURROGATE = 6
+const UTF_ERR_MISSING_SURROGATE = 7
+const UTF_ERR_INVALID = 8
+const UTF_ERR_SURROGATE = 9
+const UTF_ERR_NULL_TERMINATE = 10
+const UTF_ERR_MAX = 10
+
+type UnicodeError <: Exception
+    msg::AbstractString
+end
```

Comment:
I absolutely love enums, and tried hard to move enum support early enough in loading base so I could use it for this, but... after an hour or so of trying to move enum and doc support up, I realized it was a waste of my time for the present... and just left the doc strings bracketed with `#=` `=#`,
and changed the enum to consts.